libyara/scanner.c

```diff
@@ -498,6 +508,17 @@ YR_API int yr_scanner_scan_mem_blocks(
     if (result != ERROR_SUCCESS)
       goto _exit;
 
+    for (i = 0; i < scanner->rules->num_rules; i++)
```

Comment:
My only concern is this loop. The loop gets executed with every scanned file, and it may take a while when you have hundreds of thousands of rules as we have in VirusTotal. I think this could be donde more efficiently if we store the original bitmask in `scanner->rules`, and with every scanned file `rule_evaluate_condition_flags` is initialized by making a copy of the original bitmask using `memcpy`. This should be much more efficient as memcpy will copy multiple bits per CPU instruction, instead of having multiple instructions per bit, like in this case.

Reply:
Done